rpcs/ledgerService.go

```diff
@@ -17,89 +17,62 @@
-const ledgerServerMaxBodyLength = 512                                               // we don't really pass meaningful content here, so 512 bytes should be a safe limit
-const ledgerServerCatchupRequestBufferSize = 10
+// LedgerResponseContentType is the HTTP Content-Type header for a raw ledger block
+const LedgerResponseContentType = "application/x-algorand-ledger-v2.1"
 
-// LedgerService represents the Ledger RPC API
-type LedgerService struct {
-	ledger      *data.Ledger
-	genesisID   string
-	catchupReqs chan network.IncomingMessage
-	stop        chan struct{}
-}
+const ledgerServerMaxBodyLength = 512 // we don't really pass meaningful content here, so 512 bytes should be a safe limit
+//const blockServerCatchupRequestBufferSize = 10
 
-// EncodedBlockCert defines how GetBlockBytes encodes a block and its certificate
-type EncodedBlockCert struct {
-	_struct struct{} `codec:""`
+// LedgerServiceLedgerPath is the path to register LedgerService as a handler for when using gorilla/mux
+// e.g. .Handle(LedgerServiceLedgerPath, &ls)
+const LedgerServiceLedgerPath = "/v{version:[0-9.]+}/{genesisID}/ledger/{round:[0-9a-z]+}"
 
-	Block       bookkeeping.Block     `codec:"block"`
-	Certificate agreement.Certificate `codec:"cert"`
-}
-
-// PreEncodedBlockCert defines how GetBlockBytes encodes a block and its certificate,
-// using a pre-encoded Block and Certificate in msgpack format.
-//msgp:ignore PreEncodedBlockCert
-type PreEncodedBlockCert struct {
-	Block       codec.Raw `codec:"block"`
-	Certificate codec.Raw `codec:"cert"`
+// LedgerService represents the Ledger RPC API
+type LedgerService struct {
+	ledger        *data.Ledger
+	genesisID     string
+	net           network.GossipNode
+	enableService bool
 }
 
-// RegisterLedgerService creates a LedgerService around the provider Ledger and registers it for RPC with the provided Registrar
-func RegisterLedgerService(config config.Local, ledger *data.Ledger, registrar Registrar, genesisID string) *LedgerService {
-	service := &LedgerService{ledger: ledger, genesisID: genesisID}
-	registrar.RegisterHTTPHandler(LedgerServiceBlockPath, service)
-	c := make(chan network.IncomingMessage, config.CatchupParallelBlocks*ledgerServerCatchupRequestBufferSize)
-
-	handlers := []network.TaggedMessageHandler{
-		{Tag: protocol.UniCatchupReqTag, MessageHandler: network.HandlerFunc(service.processIncomingMessage)},
-		{Tag: protocol.UniEnsBlockReqTag, MessageHandler: network.HandlerFunc(service.processIncomingMessage)},
+// MakeLedgerService creates a LedgerService around the provider Ledger and registers it with the HTTP router
+func MakeLedgerService(config config.Local, ledger *data.Ledger, net network.GossipNode, genesisID string) *LedgerService {
+	service := &LedgerService{
+		ledger:        ledger,
+		genesisID:     genesisID,
+		net:           net,
+		enableService: config.EnableLedgerService,
+	}
+	if service.enableService {
+		net.RegisterHTTPHandler(LedgerServiceLedgerPath, service)
```

Comment:
Is it registered for: ?b={round}&v={version}

Reply:
I don't think that it need to be registered for query arguments.